Refactor weighted average aggregation page (#9961)
opensearch-project/documentation-website: _aggregations/metric/weighted-avg.md

title: Weighted average
parent: Metric aggregations
nav_order: 150
has_math: true
---

# Weighted average

The `weighted_avg` aggregation calculates the weighted average of numeric values across documents. This is useful when you want to calculate an average but weight some data points more heavily than others.

> Weighted average is calculated using the formula $$ \frac{\sum_{i=1}^n \text{value}_i \cdot \text{weight}_i}{\sum_{i=1}^n \text{weight}_i} $$.

Reviewer: ```suggestion
The weighted average is calculated using the formula $$ \frac{\sum_{i=1}^n \text{value}_i \cdot \text{weight}_i}{\sum_{i=1}^n \text{weight}_i} $$.
```

        },
        "weight": {
          "field": "num_reviews"
        },
        "format": "#.##"
      }
    }
  }
}
```
{% include copy-curl.html %}

## Example response

> The response contains the `weighted_rating`, calculated as `weighted_avg = (4.5 * 100 + 3.8 * 50) / (100 + 50) = 4.27`. Only documents 1 and 2 that contain a value for both `rating` and `num_reviews` are considered:

Reviewer: ```suggestion
The response contains the `weighted_rating`, calculated as `weighted_avg = (4.5 * 100 + 3.8 * 50) / (100 + 50) = 4.27`. Only documents 1 and 2, which contain values for both `rating` and `num_reviews`, are considered:
```

```json
{
  "took": 18,
  "timed_out": false,
  "terminated_early": true,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 3,
      "relation": "eq"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "weighted_rating": {
      "value": 4.266666650772095,
      "value_as_string": "4.27"
    }
  }
}
```

## Missing values

The `missing` parameter allows you to specify default values for documents missing the `value` field or the `weight` field instead of excluding them from the calculation.

> For example, to consider products without ratings "average", assign them a rating of 3.0 and set the `num_reviews` to 1 to give them a small non-zero weight:

Reviewer: ```suggestion
For example, to specify products without ratings as "average", assign them a rating of 3.0 and set the `num_reviews` to 1 to give them a small non-zero weight:
```
